libibverbs/verbs.h

```diff
@@ -241,9 +241,14 @@ enum ibv_rx_hash_fields {
 	IBV_RX_HASH_SRC_PORT_TCP	= 1 << 4,
 	IBV_RX_HASH_DST_PORT_TCP	= 1 << 5,
 	IBV_RX_HASH_SRC_PORT_UDP	= 1 << 6,
-	IBV_RX_HASH_DST_PORT_UDP	= 1 << 7
+	IBV_RX_HASH_DST_PORT_UDP	= 1 << 7,
 };
 
+/* In some systems/compilers enum range might be limited
+ * to signed 4 bytes.
+ */
+#define IBV_RX_HASH_INNER (1ULL << 31)
+
```

Comment:
Why not put this in the enum? It just needs to be (1UL << 31)

Reply:
The concern was that enum is signed 4 bytes, we suspect that using 1UL << 31 as part of the enum might bring some warning in some compiler/setup. If you think that this is safe we can give that a try and add it.

Reply:
We went over this in the kernel once. 

https://www.spinics.net/lists/linux-rdma/msg36828.html

The issue with this construction has always been that (1<<31) can be *negative* since '1' is technically signed, and 1 << 31 can be a signed 32 bit value. This causes the same problems as last weeks thing with comp_mask..

The only thing needed to make this safe is to make the shift unsigned:
 
  IBV_RX_HASH_INNER = 1U << 31,

Shifts over 31 will need 1ULL to avoid 32 overflow.